source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
 
         return view;
     }
+
+    public static boolean tryParseProtocolHandlerArguments(final String[] applicationArgs) {
+
+        if (applicationArgs == null) {
+            return false;
+        }
+
+        boolean found = false;
+
+        for (final String arg : applicationArgs) {
+            if (found) {
+                final URI uri = URIUtils.newURI(arg);
+
+                if (!uri.getScheme().equalsIgnoreCase(PROTOCOL_HANDLER_SCHEME)) {
+                    found = false;
+                    continue;
+                }
+
+                boolean repoUrlAvailable = false;
+                final String[] queryItems = uri.getQuery().split("&"); //$NON-NLS-1$
+
+                for (final String queryItem : queryItems) {
+                    if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_URL_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_URL_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_URL_PROPERTY, value);
+                        repoUrlAvailable = tryParseGitRepoUrl(value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_ENCODING_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_ENCODING_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_ENCODING_PROPERTY, value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_BRANCH_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_BRANCH_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_BRANCH_PROPERTY, value);
+                    }
+                }
+
+                if (repoUrlAvailable) {
+                    return true;
+                } else {
+                    found = false;
```

Comment:
It seems like you are expecting more than one -openuri and you are taking the first one that has the URL param. If you only expect one -openuri, then is seems like you should return or break here.